dm_regional_app/forms.py

```diff
@@ -325,3 +325,124 @@ class DataSourceUploadForm(forms.Form):
             )
         ],
     )
+
+
+class DynamicRateForm(forms.Form):
+    def __init__(self, *args, **kwargs):
+        self.dataframe = kwargs.pop("dataframe", None)
+        initial_data = kwargs.pop("initial_data", pd.DataFrame)
+
+        super(DynamicRateForm, self).__init__(*args, **kwargs)
+        self.initialize_fields(initial_data)
+
+    def initialize_fields(self, initial_data):
+        """
+        Dynamically create two fields for each option: one for multiplication and one for addition.
+        """
+        for index in self.dataframe.index:
+            multiply_field_name = f"multiply_{index}"
+            add_field_name = f"add_{index}"
+
+            initial_multiply = None
+            initial_add = None
+
+            # Set initial values if available, otherwise leave as None
+            if initial_data is not None:
+                try:
+                    initial_multiply = initial_data.loc[index, "multiply_value"]
+                except (KeyError, IndexError, ValueError):
+                    pass
+
+                try:
+                    initial_add = initial_data.loc[index, "add_value"]
+                except (KeyError, IndexError, ValueError):
+                    pass
```

Comment:
It's the exact same functionality, but I wonder if this would be a bit more intuitive for devs new to the code
```suggestion
            # Set initial values if available
            if initial_data is not None:
                try:
                    initial_multiply = initial_data.loc[index, "multiply_value"]
                except (KeyError, IndexError, ValueError):
                    initial_multiply = None

                try:
                    initial_add = initial_data.loc[index, "add_value"]
                except (KeyError, IndexError, ValueError):
                    initial_add = None
```